nautobot/ipam/tests/test_filters.py

```diff
@@ -1249,3 +1254,281 @@ def test_virtual_machine(self):
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
         params = {"virtual_machine": [vms[0].name, vms[1].name]}
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
+
+
+class IPAddressFilterTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+
+        IPAddress.objects.create(address="10.0.0.1/24", vrf=None, tenant=None)
+        IPAddress.objects.create(address="10.0.0.2/24", vrf=None, tenant=None)
+        IPAddress.objects.create(address="10.0.0.3/24", vrf=None, tenant=None)
+        IPAddress.objects.create(address="10.0.0.4/24", vrf=None, tenant=None)
+        IPAddress.objects.create(address="10.0.0.1/25", vrf=None, tenant=None)
+        IPAddress.objects.create(address="2001:db8::1/64", vrf=None, tenant=None)
+        IPAddress.objects.create(address="2001:db8::2/64", vrf=None, tenant=None)
+        IPAddress.objects.create(address="2001:db8::3/64", vrf=None, tenant=None)
+
+    def test_family(self):
+        self.assertEqual(IPAddress.objects.filter(host__family=4).count(), 5)
+        self.assertEqual(IPAddress.objects.filter(host__family=6).count(), 3)
+
+    def test_net_host(self):
+        self.assertEqual(IPAddress.objects.filter(host__net_host="10.0.0.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__net_host="10.0.0.2").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__net_host="10.0.0.50").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__net_host="2001:db8::1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__net_host="2001:db8::5").count(), 0)
+
+    def test_net_host_contained(self):
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="10.0.0.0/24").count(), 5)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="10.0.0.0/30").count(), 4)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="10.0.0.0/31").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="10.0.0.2/31").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="10.0.10.0/24").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="2001:db8::/64").count(), 3)
+        self.assertEqual(IPAddress.objects.filter(host__net_host_contained="2222:db8::/64").count(), 0)
+
+    def test_net_in(self):
+        self.assertEqual(IPAddress.objects.filter(host__net_in=["10.0.0.0/31", "10.0.0.2/31"]).count(), 4)
+        self.assertEqual(IPAddress.objects.filter(host__net_in=["10.0.0.0/24"]).count(), 5)
+        self.assertEqual(IPAddress.objects.filter(host__net_in=["172.16.0.0/24"]).count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__net_in=["2001:db8::/64"]).count(), 3)
+        self.assertEqual(IPAddress.objects.filter(host__net_in=["10.0.0.0/24", "2001:db8::/64"]).count(), 8)
```

Comment:
NetBox is lacking in any detailed documentation in this area, but it *looks* to me like their implementation of `net_in` supports both networks and host addresses as inputs - see https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/filtersets.py#L567 (the only usage of this custom lookup I can find) and https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/tests/test_filtersets.py#L715 (tests for that function)

Reply:
I am not sure I understand the "why" here. I feared that may be the case, but I am not sure I understand why. IMHO, it falls into
* There is a technical challenge into supporting this, as there is no network field on host, and supporting dynamic creation is difficult 
* I personally do not agree with such a design. To me, it leads to unexpected results. As an example, sending an IP without a network evaluates as a /32, which I think most would expect to sanely fail. 

Can I document the difference and keep as is?